smarty/sampler.py

```diff
@@ -407,23 +429,39 @@ def AddBetaSubstituentAtom(self, atom1type, bondset, atom2type):
         an exception if a beta substituent is attempted to be added to an atom1type which does not have an alpha 
         substituent.
         """
-        s = atom1type[0][atom1type[0].find("(")+1:atom1type[0].find(")")]
-        print s
+        #s = atom1type[0][atom1type[0].find("(")+1:atom1type[0].find(")")]
+        #if self.verbose: print s
+
+        # counting '[' tells us how many atoms are in the mix
+        count = atom1type[0].count('[')
         proposed_atomtype = ""
         number_brackets = 0
-        if s.count('[') == 1 or s.count('[') == 2:
-            # Has only one atomtype, add after the first one
-            for i in atom1type[0]:
-                proposed_atomtype += i
-                if i == ']' and number_brackets == 0:
-                    proposed_atomtype += bondset[0] + atom2type[0]
-                    number_brackets += 1
+        # find closed alpha atom
+        closeAlpha = atom1type[0].find(']')
+
+        # This has two atoms (already has an alpha atom)
+        if count == 2: 
+            proposed_atomtype = atom1type[0][:closeAlpha+1]
+            proposed_atomtype += bondset[0] + atom2type[0] + ')]'
+            #for i in atom1type[0]:
+            #    proposed_atomtype += i
+            #    if i == ']' and number_brackets == 0:
+            #        proposed_atomtype += bondset[0] + atom2type[0]
+            #        number_brackets += 1
             proposed_typename = atom1type[1] + bondset[1] + atom2type[1]
-        else:
-            # Has 3 atomtypes, add ?????
-            proposed_atomtype = '[' + atom1type[0] + bondset[0] + atom2type[0] + ']'
+            if self.verbose: print("ADD FIRST BETA SUB: proposed --- %s %s" % ( str(proposed_atomtype), str(proposed_typename)))
+
+        elif count > 2:
+            # Has an alpha atom with at least 1 beta atom
+            proposed_atomtype = atom1type[0][:closeAlpha+1]
+            proposed_atomtype += '(' + bondset[0] + atom2type[0] + ')'
+            proposed_atomtype += atom1type[0][closeAlpha+1:]
             proposed_typename = atom1type[1] + ' (' + bondset[1] + ' ' + atom2type[1] + ')'
-        print "ADD BETA SUB: proposed --- " + str(proposed_atomtype) + " " + str(proposed_typename)
+            if self.verbose: print("ADD MORE BETA SUB: proposed --- %s %s" % ( str(proposed_atomtype), str(proposed_typename)))
+
+        else:
```

Comment:
If it doesn't have an alpha atom it should add an alpha atom instead, this method shouldn't be called in that situation, but I wanted a double check included.